libs/pika/async_cuda/tests/unit/then_with_stream.cu

```diff
@@ -440,15 +448,22 @@ int pika_main()
         PIKA_TEST_EQ(dummy::host_void_calls.load(), std::size_t(0));
         PIKA_TEST_EQ(dummy::stream_void_calls.load(), std::size_t(0));
         PIKA_TEST_EQ(dummy::cublas_void_calls.load(), std::size_t(0));
+#if !defined(PIKA_HAVE_HIP)
         PIKA_TEST_EQ(dummy::cusolver_void_calls.load(), std::size_t(0));
+#endif
         PIKA_TEST_EQ(dummy::host_int_calls.load(), std::size_t(2));
         PIKA_TEST_EQ(dummy::stream_int_calls.load(), std::size_t(0));
         PIKA_TEST_EQ(dummy::cublas_int_calls.load(), std::size_t(1));
+#if !defined(PIKA_HAVE_HIP)
         PIKA_TEST_EQ(dummy::cusolver_int_calls.load(), std::size_t(0));
+#endif
+        PIKA_TEST_EQ(dummy::host_int_calls.load(), std::size_t(2));
         PIKA_TEST_EQ(dummy::host_double_calls.load(), std::size_t(0));
         PIKA_TEST_EQ(dummy::stream_double_calls.load(), std::size_t(1));
         PIKA_TEST_EQ(dummy::cublas_double_calls.load(), std::size_t(0));
+#if !defined(PIKA_HAVE_HIP)
         PIKA_TEST_EQ(dummy::cusolver_double_calls.load(), std::size_t(1));
+#endif
```

Comment:
As already mentioned, let's wait for #67 to go in before this. It's not sufficient to just remove `cusolver_*_calls` checks. The counts from those will be distributed among the `host_*_calls`, `stream_*_calls`, `cublas_*_calls` variables.

Reply:
I updated the counts, should be good now